dev/benchmarks/macrobenchmarks/lib/src/web/recorder.dart

```diff
@@ -503,54 +490,186 @@ class _WidgetBuildRecorderHostState extends State<_WidgetBuildRecorderHost> {
 class Timeseries {
   Timeseries(this.name);
 
+  /// The label of this timeseries used for debugging and result inspection.
   final String name;
 
   /// List of all the values that have been recorded.
   ///
   /// This list has no limit.
-  final List<num> _allValues = <num>[];
-
-  /// List of values that are being used for measurement purposes.
-  ///
-  /// [average], [standardDeviation] and [noise] are all based on this list, not
-  /// the [_allValues] list.
-  final List<num> _measuredValues = <num>[];
+  final List<double> _allValues = <double>[];
 
   /// The total amount of data collected, including ones that were dropped
   /// because of the sample size limit.
   int get count => _allValues.length;
 
-  /// Computes the average value of the measured values.
-  double get average => _computeAverage(name, _measuredValues);
-
-  /// Computes the standard deviation of the measured values.
-  double get standardDeviation =>
-      _computeStandardDeviationForPopulation(name, _measuredValues);
-
-  /// Computes noise as a multiple of the [average] value.
-  ///
-  /// This value can be multiplied by 100.0 to get noise as a percentage of
-  /// the average.
+  /// Extracts useful statistics out of this timeseries.
   ///
-  /// If [average] is zero, treats the result as perfect score, returns zero.
-  double get noise => average > 0.0 ? standardDeviation / average : 0.0;
+  /// See [TimeseriesStats] for more details.
+  TimeseriesStats computeStats() {
+    // The first few values we simply discard and never look at. There's from warm-up phase.
+    final List<double> warmUpValues = _allValues.sublist(0, _allValues.length - _kMeasuredSampleCount);
+
+    // Values we analyze.
+    final List<double> candidateValues = _allValues.sublist(_allValues.length - _kMeasuredSampleCount);
+
+    // The average that includes outliers.
+    final double dirtyAverage = _computeAverage(name, candidateValues);
+
+    // The standard deviation that includes outliers.
+    final double dirtyStandardDeviation = _computeStandardDeviationForPopulation(name, candidateValues);
+
+    // Any value that's higher than this is considered an outlier.
+    final double outlierCutOff = dirtyAverage + dirtyStandardDeviation;
```

Comment:
Do we not care about the other side of outliers? That is outliers that are way below average (i.e. below `dirtyAverage - dirtyStandardDeviation`).

Reply:
One fundamental property of our benchmarks is that the metrics are always "lower is better". The assumption is that unless there's a severe bug in the benchmark, even the lowest value will contain all the work we're interested in measuring.

Conversely, outliers tend to happen due to inefficiencies that happen at unpredictable points in time, such as GC pauses, canvas reallocations, etc. This is still work we should measure as it is a strong indicator of app performance. A lot of these things cannot be fixed as they are part of the trade-offs we decided to make. One example is where we try to reuse the same canvas, but eventually that canvas won't be sufficient and we'll reallocate, paying a lot of one-time cost, then try to amortize that cost over the next N frames. This change is not removing the data contributed by the outliers, but only puts it in a separate category.

Reply:
Having said that, you just reminded me that I didn't include the outlier average in the score keys. Fixed.